src/Components/Classes/index.jsx

```diff
@@ -1,73 +1,40 @@
 import styles from './classes.module.css';
 import ClassList from './List/ClassList';
-import React, { useState, useEffect } from 'react';
-import { Link, useHistory } from 'react-router-dom/cjs/react-router-dom.min';
+import React, { useEffect } from 'react';
+import { Link } from 'react-router-dom/cjs/react-router-dom.min';
 import { useSelector, useDispatch } from 'react-redux';
 import Loader from '../Shared/Loader';
 import { getClasses } from '../../Redux/Classes/thunks';
 import Button from '../Shared/Button';
 import ResponseModal from '../Shared/ResponseModal';
+import { handleDisplayToast } from '../../Redux/Shared/ResponseToast/actions';
 
 function Projects() {
-  const [showModal, setShowModal] = useState({ show: false, msg: '', state: '' });
-  const [selectedClass, setSelectedClass] = useState(null);
   const dispatch = useDispatch();
-  const history = useHistory();
-  const classes = useSelector((state) => state.classes.data);
-  const pending = useSelector((state) => state.classes.isPending);
+  const { data, isPending } = useSelector((state) => state.classes);
+  const { show, message, state } = useSelector((state) => state.toast);
+  const { success } = useSelector((state) => state.classes);
+
+  console.log(success);
 
   useEffect(() => {
     getClasses(dispatch);
   }, [dispatch]);
 
-  useEffect(() => {
-    if (history.location.state) {
-      setShowModal({
-        show: history.location.state.show || false,
-        msg: history.location.state.msg || '',
-        state: 'success'
-      });
-    }
-    const objHistory = {
-      ...location,
-      state: { show: false, msg: '', state: '' }
-    };
-    setTimeout(() => {
-      setShowModal({ show: false, msg: '', state: '' });
-    }, 2000);
-    history.replace(objHistory);
-  }, []);
-
-  const getById = async (id) => {
-    try {
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/class/${id}`);
-      const data = await response.json();
-      setSelectedClass(data.data);
-    } catch (error) {
-      setShowModal({ show: true, state: 'fail', msg: 'Something went wrong :( try again later' });
-    }
-  };
-
   return (
     <section className={styles.container}>
       <h2>Class List</h2>
-      {pending && <Loader />}
-      {classes.length > 0 && !pending && (
-        <ClassList
-          classes={classes && classes}
-          getById={getById}
-          selectedClass={selectedClass}
-        ></ClassList>
-      )}
-      {!pending && classes.length === 0 && 'There are not classes yet. Add new ones!'}
+      {isPending && <Loader />}
+      {data.length > 0 && !isPending && <ClassList classes={data && data}></ClassList>}
```

Comment:
Yes here you are using the useSelector and passing the list as prop. As you want, I think it's okey, just saying due the problematic with this warning about passing data as props